File: Servidor/src/main/java/com/gotravel/model/ResenaId.java
package com.gotravel.model;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.Hibernate;

import java.util.Objects;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Embeddable
public class ResenaId implements java.io.Serializable {

    private static final long serialVersionUID = -3017767775810651976L;

    @Column(name = "id_usuario", nullable = false)
    private Integer idUsuario;

    @Column(name = "id_contratacion", nullable = false)
    private Integer idContratacion;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
        ResenaId entity = (ResenaId) o;
        return Objects.equals(this.idUsuario, entity.idUsuario) &&
                Objects.equals(this.idContratacion, entity.idContratacion);
    }

    @Override
    public int hashCode() {
        return Objects.hash(idUsuario, idContratacion);
    }

}

File: GoTravelMobile/app/src/main/java/com/gotravel/mobile/data/model/ResenaId.kt
package com.gotravel.mobile.data.model

import kotlinx.serialization.Serializable
import lombok.AllArgsConstructor
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.Setter
import java.util.Objects

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Serializable
data class ResenaId(
    val idUsuario: Int,
    val idServicio: Int
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || this.javaClass != other.javaClass) return false
        val entity = other as ResenaId
        return idUsuario == entity.idUsuario && idServicio == entity.idServicio
    }

    override fun hashCode(): Int {
        return Objects.hash(idUsuario, idServicio)
    }

}